modules/parquet/src/parquetjs/compression.ts

```diff
@@ -73,6 +73,14 @@ export const PARQUET_COMPRESSION_METHODS: Record<ParquetCompression, Compression
  * @param options.modules External library dependencies
  */
 export async function preloadCompressions(options?: {modules: {[key: string]: any}}) {
+  // Re-initialize compression instances with new options
+  const mergedOptions = {...options, modules: {...modules, ...options?.modules}};
+  for (const method in PARQUET_COMPRESSION_METHODS) {
+    PARQUET_COMPRESSION_METHODS[method] = new PARQUET_COMPRESSION_METHODS[method].constructor(
```

Comment:
OK for now, this is clever. But long term I feel that it is too messy, too hard to explain to users. It would probably be cleaner just to be able to just call `compression.preload({modules})` multiple times.